packages/tasks/src/tasks/chat-completion/spec/input.json

```
@@ -0,0 +1,54 @@
+{
+	"title": "ChatCompletionInput",
+	"description": "Inputs for ChatCompletion inference",
+	"type": "object",
+	"properties": {
+		"messages": {
+			"type": "array",
+			"title": "ChatCompletionInputMessage",
+			"items": {
+				"type": "object",
+				"properties": {
+					"role": {
+						"oneOf": [{ "const": "assistant" }, { "const": "system" }, { "const": "user" }],
+						"description": "The role of the messages author."
+					},
+					"content": {
+						"type": "string",
+						"description": "The content of the message."
+					}
+				},
+				"required": ["role", "content"]
+			}
+		},
+		"frequency_penalty": {
+			"type": "number",
+			"description": "Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim."
+		},
+		"max_tokens": {
+			"type": "integer",
+			"description": "The maximum number of tokens that can be generated in the chat completion."
+		},
+		"seed": {
+			"type": "integer",
+			"description": "The random sampling seed."
+		},
+		"stop": {
+			"oneOf": [{ "type": "string" }, { "type": "array", "items": { "type": "string" } }],
+			"description": "Stop generating tokens if a stop token is generated."
+		},
```

Same comment - let's maybe add a `title`
```suggestion
		"stop": {
			"oneOf": [{ "type": "string" }, { "type": "array", "items": { "type": "string" } }],
            "title": "ChatCompletionInputStopReason",
			"description": "Stop generating tokens if a stop token is generated."
		},
```

---

Could even be moved to `common-definitions` if it makes sense to share the definition with `text-generation`

---

I thought about it but Python-wise it makes things more difficult to handle. If that's fine with you, I'll prefer to keep them separated.

---

Sounds good!